### ISSUE TITLE
Editor appending closing tags by itself.

### ISSUE THREAD

**Issue by VladAlenitsev** [samclarke/SCEditor#833]
SCEditor adds closing tags which result in unwanted HTML. Possibly only relatively "newer" tags are affected, i.e `<article>`, `<section>`, `<nav>` but not `<div>`. This behaviour occurs when when switching between preview and view source modes multiple times.

For example having following code in preview
```
<div>
    <section class="title">
        <div class="title_header">
            <div class="content">
                <h1>Title</h1>
                <p>Description ...</p>
            </div>
        </div>
    </section>
</div>
```

and switching to source view, code becomes

```
<div>
    <section class="title">
        </section><div class="title_header" style="">
            <div class="content">
                <h1>Title</h1>
                <p>Description ...</p>
            </div>
        </div><section class="title">
    </section>
</div>
```

**Comment by live627** [comment]
We need a [basic reproducible case](https://stackoverflow.com/help/minimal-reproducible-example). IOW, you should provide the code that you use to initialize the editor with any custom formats.

**Comment by samclarke** [comment]
Thanks for reporting! The issue is caused by `fixNesting()` not knowing `<section>` tags are blocks.

Should be fixed by #825 in the next release.
